mlflow/types/llm.py

```diff
@@ -90,24 +127,73 @@ def from_dict(cls, data):
         return cls(**filtered_data)
 
 
+@dataclass
+class FunctionToolCallArguments(_BaseDataclass):
+    """
+    The arguments of a function tool call made by the model.
+
+    Args:
+        arguments (str): A JSON string of arguments that should be passed to the tool.
+        name (str): The name of the tool that is being called.
+    """
+
+    name: str
+    arguments: str
+
+    def __post_init__(self):
+        self._validate_field("name", str, True)
+        self._validate_field("arguments", str, True)
+
+    def to_tool_call(self, id):
```

Comment:
Why do we need this?

Reply:
I thought it was easier than users calling

```python
# without convenience method
tool_call_args = FunctionToolCallArguments(name="get_weather", arguments="")
tool_call = ToolCall(id=id, function=tool_call_args)

# with
tool_call = FunctionToolCallArguments(name="get_weather", arguments="").to_tool_call(id)
```

Eventually as OpenAI (or maybe Databricks or some other provider) adds more types of tools, this `to_tool_call()` method may be more helpful than requiring the user to remember what args `ToolCall` needs to be provided with

```python
# without the method, users need to specify `uc_function`, not `function`
tool_call_args = UnityCatalogToolCallArguments(name="some_tool", catalog="main.ml.tools")
tool_call = ToolCall(id=id, uc_function=tool_call_args)

# with method, they don't need to think
UnityCatalogToolCallArguments(name="some_tool", catalog="main.ml.tools").to_tool_call()
```